src/auth/dto/auth-register.input.ts

```diff
@@ -0,0 +1,25 @@
+import { Field, InputType } from '@nestjs/graphql';
+import {
+  IsString,
+  IsEmail,
+  MinLength,
+  MaxLength,
+  Matches,
+} from 'class-validator';
+
+@InputType()
+export class AuthRegisterInput {
+  @IsString()
+  @IsEmail()
+  @Field()
+  email: string;
+
+  @IsString()
+  @MinLength(8)
+  @MaxLength(32)
+  @Matches(/((?=.*\d)|(?=.*\W+))(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$/, {
+    message: 'password is too weak',
```

Comment:
Ja bym używał zmiennych do definiowania błędów i wysyłał na frontend kod błędu, wtedy możesz łatwo tłumaczyć na różne języki w aplikacji.

Zrobiłbym folder /commons/errors.ts

I dodał w nim:
export const ERRORS {
 E0001 = 'password is too weak'
}
A tutaj wstawił:
 message: ERRORS.E0001;